### PR TITLE
New endpoint for creation claims from backoffice

### PR DESCRIPTION
**Pull request by alex-bel**
Added new property to Claim - ClaimSource

### CODE REVIEW

File: src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.axonframework.commandhandling.TargetAggregateIdentifier;
+
+import java.time.LocalDateTime;
+
+@Value
+public class CreateBackofficeClaimCommand {
+  @TargetAggregateIdentifier
+  String id;
+  String userId;
+  LocalDateTime registrationDate;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java]: Shall we use Instant here, and translate to localdatetime in the eventlistener? when we are going to save it to DB

File: src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import java.time.LocalDateTime;
+
+@Value
+public class BackofficeClaimCreatedEvent {
+
+  @AggregateIdentifier
+  private String id;
+  private String userId;
+  private LocalDateTime registrationDate;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.java]: Same comment with the command Instant

File: src/main/java/com/hedvig/claims/web/InternalController.java [hunk 3]
```diff
@@ -82,6 +85,19 @@ public InternalController(CommandBus commandBus, ClaimsRepository repository, Cl
...
 
+  @RequestMapping(path = "/createFromBackOffice", method = RequestMethod.POST)
+  public ResponseEntity<?> createClaim(@RequestBody CreateBackofficeClaimDTO req) {
+    log.info("Claim recieved!:" + req.toString());
+    UUID uid = UUID.randomUUID();
+    commandBus.sendAndWait(
+      new CreateBackofficeClaimCommand(
+        uid.toString(),
+        req.getUserId(),
+        req.getRegistrationDate().atZone(SWEDEN_TZ).toLocalDateTime(),
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/web/InternalController.java]: same comment for instant

File: src/main/java/com/hedvig/claims/query/ClaimsEventListener.java [hunk 4]
```diff
@@ -70,6 +76,32 @@ public void on(ClaimCreatedEvent e) {
     claimRepository.save(claim);
   }
 
+  @EventHandler
+  public void on(BackofficeClaimCreatedEvent e) {
+    log.info("BackofficeClaimCreatedEvent: " + e);
+    ClaimEntity claim = new ClaimEntity();
+    claim.id = e.getId();
+    claim.userId = e.getUserId();
+    claim.registrationDate = e.getRegistrationDate();
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/query/ClaimsEventListener.java]: so basically here, we can convert it to LocalDateTime

File: src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java (hunk 1 above)
Comment (alex-bel) [reply src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java]: Claim's registrationDate is now LocalDateTime and refactoring it is not in the scope of the ticket I believe

File: src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+
+@Value
+public class BackofficeClaimCreatedEvent {
+
+  @AggregateIdentifier
+  private String id;
+  private String userId;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/events/BackofficeClaimCreatedEvent.java]: This refers to members, so its better to rename it to `memberId`

File: src/main/java/com/hedvig/claims/web/dto/CreateBackofficeClaimDTO.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package com.hedvig.claims.web.dto;
+
+import com.hedvig.claims.aggregates.ClaimSource;
+import lombok.Data;
+
+import java.time.Instant;
+
+@Data
+public class CreateBackofficeClaimDTO {
+  String userId;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/web/dto/CreateBackofficeClaimDTO.java]: same memberId

File: src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import org.axonframework.commandhandling.TargetAggregateIdentifier;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+
+@Value
+public class CreateBackofficeClaimCommand {
+  @TargetAggregateIdentifier
+  String id;
+  String userId;
```
Comment (meletios) [inline src/main/java/com/hedvig/claims/commands/CreateBackofficeClaimCommand.java]: same memberId

**Pull request merged**